src/api/create-vis.js

```diff
@@ -53,23 +54,37 @@ var loadVizJSON = function (el, visModel, vizjsonData, options) {
   var showLegends = true;
   if (_.isBoolean(options.legends)) {
     showLegends = options.legends;
-  } else if (_.isBoolean(vizjson.legends)) {
-    showLegends = vizjson.legends;
+  } else if (vizjson.embed_options && _.isBoolean(vizjson.embed_options.legends)) {
+    showLegends = vizjson.embed_options.legends;
   }
 
+  var showLayerSelector = true;
+  if (_.isBoolean(options.layer_selector)) {
+    showLayerSelector = options.layer_selector;
+  } else if (vizjson.embed_options && _.isBoolean(vizjson.embed_options.layer_selector)) {
+    showLayerSelector = vizjson.embed_options.layer_selector;
+  }
+
+  var settingsModel = new SettingsModel({
+    showLegends: showLegends,
+    showLayerSelector: showLayerSelector
+  });
+
   visModel.set({
     title: options.title || vizjson.title,
     description: options.description || vizjson.description,
     apiKey: options.apiKey,
     authToken: options.authToken,
-    showLegends: showLegends,
     showEmptyInfowindowFields: options.show_empty_infowindow_fields === true,
     https: isProtocolHTTPs || options.https === true || vizjson.https === true
   });
 
+  visModel.settings = settingsModel;
```

Comment:
This kind of strange... the visModel won't have a `settings` prop until we get to this point. Imagine the following scenario:

```
var vis = createVis(...);
vis.settings.set('showLegends', true);
```
Here, we'll get an error because `vis.settings` is still `undefined`... right?

Idea:

```
// visModel
...
initialize: () {
  ...
  // A vis model gets some settings (with defaults)
  this.settings = new Settings();
},

...
// we provide a method to set some new settings
setSettings: function (settings) {
  this.settings.set(settings);
}
...
```
in `create-vis.js`, once the vizjson has been loaded (instead of this) we can do:

```
visModel.setSettings({...});
```